plugin-server/src/worker/ingestion/person-state.ts

```diff
@@ -706,13 +706,142 @@ export class PersonOverrideWriter {
 
         return id
     }
+
+    public async getPersonOverrides(teamId: number): Promise<PersonOverrideDetails[]> {
+        const { rows } = await this.postgres.query(
+            PostgresUse.COMMON_WRITE,
+            SQL`
+                SELECT
+                    override.team_id,
+                    old_person.uuid as old_person_id,
+                    override_person.uuid as override_person_id,
+                    oldest_event
+                FROM posthog_personoverride override
+                LEFT OUTER JOIN posthog_personoverridemapping old_person
+                    ON override.team_id = old_person.team_id AND override.old_person_id = old_person.id
+                LEFT OUTER JOIN posthog_personoverridemapping override_person
+                    ON override.team_id = override_person.team_id AND override.override_person_id = override_person.id
+                WHERE override.team_id = ${teamId}
+            `,
+            undefined,
+            'getPersonOverrides'
+        )
+        return rows.map((row) => ({
+            ...row,
+            oldest_event: DateTime.fromISO(row.oldest_event),
+        }))
```

Comment:
This method was pulled in from tests so that I could swap the entire backing implementation at once without having to coordinate it across multiple abstractions. This method is really only intended to be used in tests, though.

(I did rewrite the query here because I found [the other one](https://github.com/PostHog/posthog/blob/51dd5d78c93e393ef87a996d9b572423f3e8adc3/plugin-server/tests/worker/ingestion/person-state.test.ts#L31-L51) challenging to read, though looking back at it now, I do think that the CTE having the predicate associated with it versus being specified here 3x is a nice idea.)

Reply:
Makes sense to me. I have a minor preference for naming things like `fooForTests` or something to avoid footguns. It'd be pretty weird for someone to reach for this... but...